src/main/scala/firrtl/passes/RemoveCHIRRTL.scala

```diff
@@ -118,7 +119,7 @@ object RemoveCHIRRTL extends Pass {
         set_write(rws, "wdata", "wmask")
       val mem = DefMemory(s.info, s.name, s.tpe, s.size, 1, if (s.seq) 1 else 0,
                   rds map (_.name), wrs map (_.name), rws map (_.name))
-      Block(mem +: stmts)
+      if (stmts.nonEmpty) Block(mem +: stmts) else EmptyStmt
```

Comment:
I wasn't saying this is the wrong thing to do; I was just trying to understand it.

Note that other passes still might have to understand Mems with no ports, though!  Future optimizations could delete the ports.